pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -259,7 +259,7 @@ public HelixAdmin getHelixAdmin() {
   private void addInstanceGroupTagIfNeeded() {
     InstanceConfig instanceConfig = getHelixInstanceConfig(_instanceId);
     assert instanceConfig != null;
-    if (!instanceConfig.containsTag(Helix.CONTROLLER_INSTANCE)) {
+    if (instanceConfig != null && !instanceConfig.containsTag(Helix.CONTROLLER_INSTANCE)) {
```

Comment:
We have an assert right before this statement. Why check again?

Reply:
I believe the assert will trigger only with asserts enabled, but it appears that it's normal for instanceConfig to be null when we boot the benchmark tool. There might be another fix here, to make the instanceConfig present, but from what I can see, we launch the helix manager twice and it's the second launch that's missing the config.

Reply:
@grcevski, sorry for the delay,  can you post the stack trace please?

Reply:
Hey @mcvsubbu, no problem at all, I was away as well for the holidays :). Here's a stack trace we see (apologies for the formatting) :
Exception caught:
java.lang.NullPointerException: null
        at org.apache.pinot.controller.helix.core.PinotHelixResourceManager.addInstanceGroupTagIfNeeded(PinotHelixResourceManager.java:262) ~[pinot-controller-0.3.0-SNAPSHOT.jar:0.3.0-SNAPSHOT-63ceb0c47759cf5b7372c03a2d76f7206bbec133]
        at org.apache.pinot.controller.helix.core.PinotHelixResourceManager.start(PinotHelixResourceManager.java:189) ~[pinot-controller-0.3.0-SNAPSHOT.jar:0.3.0-SNAPSHOT-63ceb0c47759cf5b7372c03a2d76f7206bbec133]
        at org.apache.pinot.tools.perf.PerfBenchmarkDriver.startHelixResourceManager(PerfBenchmarkDriver.java:258) ~[pinot-tools-0.3.0-SNAPSHOT.jar:0.3.0-SNAPSHOT-63ceb0c47759cf5b7372c03a2d76f7206bbec133]
        at org.apache.pinot.tools.perf.PerfBenchmarkDriver.run(PerfBenchmarkDriver.java:172) ~[pinot-tools-0.3.0-SNAPSHOT.jar:0.3.0-SNAPSHOT-63ceb0c47759cf5b7372c03a2d76f7206bbec133]
        at org.apache.pinot.tools.perf.PerfBenchmarkRunner.startServerWithPreLoadedSegments(PerfBenchmarkRunner.java:122) ~[pinot-tools-0.3.0-SNAPSHOT.jar:0.3.0-SNAPSHOT-63ceb0c47759cf5b7372c03a2d76f7206bbec133]
        at org.apache.pinot.tools.perf.PerfBenchmarkRunner.execute(PerfBenchmarkRunner.java:99) ~[pinot-tools-0.3.0-SNAPSHOT.jar:0.3.0-SNAPSHOT-63ceb0c47759cf5b7372c03a2d76f7206bbec133]
        at org.apache.pinot.tools.PinotToolLauncher.execute(PinotToolLauncher.java:58) ~[pinot-tools-0.3.0-SNAPSHOT.jar:0.3.0-SNAPSHOT-63ceb0c47759cf5b7372c03a2d76f7206bbec133]
        at org.apache.pinot.tools.PinotToolLauncher.main(PinotToolLauncher.java:69) ~[pinot-tools-0.3.0-SNAPSHOT.jar:0.3.0-SNAPSHOT-63ceb0c47759cf5b7372c03a2d76f7206bbec133]

The command line I'm running the benchmark server with is:

pinot-tools.sh PerfBenchmarkRunner -mode startAll -dataDir temp/optimal_startree_small_yearly -tableNames tpch_lineitem_OFFLINE

Thanks a bunch!

Reply:
Hello @mcvsubbu , just checking-in to see if you had time to review the stack trace. Thanks!

Reply:
Yes, I finally got around to it, and my sincere apologies for the delay.

Please remove the assert statement before the line. Instead, add a comment that instanceConfig can be null when connecting as a participant while running from PerfBenchmarkRunner.

Also, another minor fix is needed.

Please change line 241 in PerfBenchmarkDriver to set the correct instance name for the server. It should be
`serverConfiguration.setProperty(CommonConstants.Server.CONFIG_OF_INSTANCE_ID, _serverInstanceName);`

Thanks once again for the fix.

Reply:
Thank you for the review and the suggestions @mcvsubbu! I have added the comment and fixed the server name property in PerfBenchmarkDriver. I rebased and tested the benchmark with the changes locally to make sure everything is still in order.

Reply:
Sorry again about the rebase issue @mcvsubbu. I've cleaned up the branch and it correctly shows the 3 changes. Thanks!